salt/client/ssh/shell.py

```diff
@@ -31,22 +32,15 @@
 RSTR_RE = re.compile(r"(?:^|\r?\n)" + RSTR + r"(?:\r?\n|$)")
 
 
-class NoPasswdError(Exception):
-    pass
-
-
-class KeyAcceptError(Exception):
-    pass
-
-
 def gen_key(path):
     """
     Generate a key for use with salt-ssh
     """
-    cmd = 'ssh-keygen -P "" -f {0} -t rsa -q'.format(path)
-    if not os.path.isdir(os.path.dirname(path)):
+    cmd = ["ssh-keygen", "-P", '""', "-f", path, "-t", "rsa", "-q"]
```

Comment:
This changes the passphrase from empty string to a string containing quotes. Is it really intended?

Reply:
Just to explain why this happens: `cmd` always contained `-P ""`, the difference is that it was passed to a shell which looked at `""` and saw it as an empty string. Now that the `subprocess.call` is without `shell=True`, nothing is in between this line and `ssh-keygen` that could strip the `""`.  
To fix this, it should be enough to remove the `'""'` list element.

Reply:
Agreed, I will update this.